tests/ignite/metrics/test_psnr.py

```diff
@@ -0,0 +1,236 @@
+import os
+
+import numpy as np
+import pytest
+import torch
+from skimage.metrics import peak_signal_noise_ratio as ski_psnr
+
+import ignite.distributed as idist
+from ignite.exceptions import NotComputableError
+from ignite.metrics import PSNR
+from ignite.utils import manual_seed
+
+
+def test_zero_div():
+    psnr = PSNR()
+    with pytest.raises(NotComputableError, match="PSNR must have at least one example before it can be computed"):
+        psnr.compute()
+
+
+def test_invalid_psnr():
+    y_pred = torch.rand(1, 3, 8, 8)
+    y = torch.rand(1, 3, 8, 8)
+
+    with pytest.raises(TypeError, match=r"Expected y_pred and y to have the same data type."):
+        psnr = PSNR()
+        psnr.update((y_pred, y.double()))
+
+    with pytest.raises(ValueError, match=r"Expected y_pred and y to have the same shape."):
+        psnr = PSNR()
+        psnr.update((y_pred, y.squeeze(dim=0)))
+
+    with pytest.raises(ValueError, match=r"y has intensity values outside the range expected"):
+        psnr = PSNR()
+        psnr.update((y_pred, y))
+        # to catch ValueError for this batch
+        psnr.update((y_pred, y + 1.0))
+
+
+def _test_psnr(y_pred, y, data_range, device):
```

Comment:
Can we add a consistency test batch result vs image-wise computation ?

Reply:
On it!